ts/src/pro/toobit.ts

```diff
@@ -0,0 +1,563 @@
+    }
+
+    parseWsOHLCV (ohlcv, market = undefined): OHLCV {
+        //
+        //             {
+        //                 t: 1757251200000,
+        //                 o: '0.21897',
+        //                 h: '0.21898',
+        //                 l: '0.21889',
+        //                 c: '0.21889',
+        //                 v: '5247',
+        //                 s: 'DOGEUSDT',
+        //                 sn: 'DOGEUSDT',
+        //                 st: 0
+        //             }
+        //
+        const parsed = this.parseOHLCV (ohlcv, market);
+        return parsed;
+    }
+
+    /**
+     * @method
+     * @name toobit#watchTicker
+     * @see https://toobit-docs.github.io/apidocs/spot/v1/en/#individual-symbol-ticker-streams
+     * @description watches a price ticker, a statistical calculation with the information calculated over the past 24 hours for a specific market
+     * @param {string} symbol unified symbol of the market to fetch the ticker for
+     * @param {object} [params] extra parameters specific to the exchange API endpoint
+     * @returns {object} a [ticker structure]{@link https://docs.ccxt.com/#/?id=ticker-structure}
+     */
+    async watchTicker (symbol: string, params = {}): Promise<Ticker> {
+        await this.loadMarkets ();
+        symbol = this.symbol (symbol);
+        const tickers = await this.watchTickers ([ symbol ], params);
+        return tickers[symbol];
+    }
+
+    /**
+     * @method
+     * @name toobit#watchTickers
+     * @see https://toobit-docs.github.io/apidocs/spot/v1/en/#individual-symbol-ticker-streams
+     * @description watches a price ticker, a statistical calculation with the information calculated over the past 24 hours for all markets of a specific list
+     * @param {string[]} symbols unified symbol of the market to fetch the ticker for
+     * @param {object} [params] extra parameters specific to the exchange API endpoint
+     * @returns {object} a [ticker structure]{@link https://docs.ccxt.com/#/?id=ticker-structure}
+     */
+    async watchTickers (symbols: Strings = undefined, params = {}): Promise<Tickers> {
+        await this.loadMarkets ();
+        if (symbols === undefined) {
+            throw new ArgumentsRequired (this.id + ' watchTradesForSymbols() requires a non-empty array of symbols');
+        }
+        symbols = this.marketSymbols (symbols);
+        // const streamHash = 'multipleTrades' + '::' + symbols.join (',');
+        // const firstMarket = this.market (symbols[0]);
+        const messageHashes = [];
+        const subParams = [];
+        for (let i = 0; i < symbols.length; i++) {
+            const symbol = symbols[i];
+            const market = this.market (symbol);
+            messageHashes.push ('ticker::' + symbol);
+            const rawHash = market['id'];
+            subParams.push (rawHash);
+        }
+        const marketIds = this.marketIds (symbols);
+        const url = this.urls['api']['ws']['spot'] + '/quote/ws/v1';
```

Comment:
they don't have swap streams?

Reply:
yeah, i'll just rename that key to 'common' (instead of spot/swap)